build-scripts/component-common.gradle

```diff
@@ -0,0 +1,73 @@
+// This script defines common setup logic for our components, such as depending
+// on the correct versions of android dependencies.
+// Absent some special need for customization, we expect each project under `/components`
+// to apply this script to their build process via:
+//
+// ```
+// apply from: "$rootDir/build-scripts/component-common.gradle"
+// ```
+
+
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
+
+android {
+    ndkVersion rootProject.ext.build.ndkVersion
+    compileSdkVersion rootProject.ext.build.compileSdkVersion
+
+    defaultConfig {
+        minSdkVersion rootProject.ext.build['minSdkVersion']
+        targetSdkVersion rootProject.ext.build['targetSdkVersion']
+
+        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+        buildConfigField("String", "LIBRARY_VERSION", "\"${rootProject.ext.library.version}\"")
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+            consumerProguardFiles "$rootDir/proguard-rules-consumer-jna.pro"
+        }
+    }
+
+    // This is required to support new AndroidX support libraries.
+    // See mozilla-mobile/android-components#842
+    compileOptions {
+        sourceCompatibility rootProject.ext.build.jvmTargetCompatibility
+        targetCompatibility rootProject.ext.build.jvmTargetCompatibility
+    }
+
+    // We're told the following is required to support new AndroidX support libraries.
+    // See mozilla-mobile/android-components#842
+    //
+    // tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
+    //    kotlinOptions {
+    //         jvmTarget = rootProject.ext.build.jvmTargetCompatibility
+    //     }
+    // }
+    //
+    // However, it doesn't work when factored out into this script.
```

Comment:
Specifically, if I try to do the previous snippet in this build script, gradle tries to look up `org.jetbrains....` as a property on `$project`, which fails. Maybe there's some trick I don't know about to have it looked up as a global package name?

Reply:
More concerningly, I don't know how to test whether this change has broken whatever AndroidX thing it was introduced to address; our `./gradlew test` still passes and an a-c test run seems to be going OK so far. I'll try a Fenix build with locally-published components as well for completeness.

Reply:
> I'll try a Fenix build with locally-published components as well for completeness.

I can't build Fenix against latest a-c right now, looks like there might be a refactor in progress that needs updates in Fenix. Will try again later, but definitely want to do a full test run on Fenix before merging this.

Reply:
It took me a bit of fiddling around to find the right combo of branches, but I have successfully build Fenix against a local publish of appservices and a-c with these changes in place, and confirmed that sync- and accounts-related functionality is working as expected